rust/server-framework/src/lib.rs

```diff
@@ -55,8 +55,8 @@ where
             bail!("Probes and metrics handler unexpectedly exited");
         },
         _ = main_task_handler => {
-            error!("Main task unexpectedly exited");
```

Comment:
are you sure this is OK to remove? I remember that at one point we didn't have proper error exiting and the panics just silently failed. Did you test this?

Reply:
agreed this change seems dangerous at first glance

Reply:
This change is so the when ending_version is reached, the processor stops gracefully and doesn't log it as an error. 

Verified that panic in the fetcher thread and processor thread still work.
<img width="1426" alt="Screenshot 2024-05-17 at 11 19 37 AM" src="https://github.com/aptos-labs/aptos-indexer-processors/assets/8248583/e7013236-7b0c-45a5-81c7-b4e00696974a">
<img width="1426" alt="Screenshot 2024-05-17 at 11 09 42 AM" src="https://github.com/aptos-labs/aptos-indexer-processors/assets/8248583/7bbea0a8-2679-474f-8985-953016d00cf7">

Reply:
Can you try to crash something like in the processors and add to test plan?

Reply:
I added a panic! in the processors thread and it exited with error. Let me include the screenshots in test plan.

Reply:
I think we can do something like this (pls verify actual behavior of `JoinHandle` first), instead of throwing it away
```
        res = main_task_handler => {
            res.expect("Main task failed").expect("Main task failed");
        },
```

Reply:
dis https://docs.rs/tokio/latest/tokio/task/struct.JoinError.html